09_maps/main.go

```diff
@@ -111,18 +122,194 @@ func main() {
+
+	// var result float64 = bmi
+
+	value, exists = mapInput["height"]
+	if !exists || value == 0 {
+		fmt.Println("Height not found")
+	}
+	value, exists = mapInput["weight"]
+	if !exists || value == 0 {
+		fmt.Println("Weight not found")
+	}
+	// debug
+	// fmt.Println("weight debug", mapInput["weight"])
+	// fmt.Println("height debug", mapInput["height"])
+	// debug
+	// fmt.Println(heightAdjust)
+	// fmt.Println(weightConverted)
+	bmi = (weightConverted / (heightConverted * heightConverted))
+	bmi = bmi * 10000
+	// debug
+	// fmt.Println(bmi)
+	// TODO: check and fix syntax for name and bmi variable
+	// fmt.Println("BMI is", bmi)
+	return bmi
+
+	// for {
+	// 	var counter int = 0
+	// 	if counter > len(mapInput) {
+	// 		break
+	// 	}
+	// 	counter++
+	// 	return bmi
+	// }
+
+}
+
+// TODO: Create a function that takes an array of maps as an argument and
+// prints the average BMI of the people in the array.
+
+// create function
+// accept array of maps as input
+// loop through all arrays (e.g. "people" array of maps)
+// calculate BMI (calcBMI function)
+// add the sum of all array elements
+// increase counter
+// divide through the amount of array elements
+// print result = avg BMI
+func calcAvgBMI(arrayOfMapInput []map[string]int) float64 {
+	var counter int
+	var bmiSum float64
+	var avgBMI float64
+	var lenConvert int = len(arrayOfMapInput)
+	var lenConverted float64 = float64(lenConvert)
+	for {
+		if counter >= len(arrayOfMapInput) {
+			break
+		}
+		bmiSum += calcBMI(arrayOfMapInput[counter])
+		// debug
+		// fmt.Println(bmiSum)
+		counter++
+	}
+	// debug
+	// fmt.Println(lenConverted)
+	avgBMI = bmiSum / lenConverted
+	// debug
+	// fmt.Println("avg debug print", avgBMI)
+	return avgBMI
 }
+
+// TODO: Create a function that takes an array of maps as an argument and
+// returns the map with the highest BMI.
+
+// create function
+// accept array of maps as input
+// calculate the BMI with function calcBMI for all elements of the array
+// compare all elements with each other to get highest value/BMI
+// if one element is bigger than ALL of the others, return that element
+// else continue comparing
+
+func highestBMI(arrayOfMapInput []map[string]int) map[string]int {
+	// testing hard coded array element comparison with AND condition
+	if calcBMI(arrayOfMapInput[0]) > calcBMI(arrayOfMapInput[1]) && calcBMI(arrayOfMapInput[0]) > calcBMI(arrayOfMapInput[2]) && calcBMI(arrayOfMapInput[0]) > calcBMI(arrayOfMapInput[3]) && calcBMI(arrayOfMapInput[0]) > calcBMI(arrayOfMapInput[4]) {
+		return arrayOfMapInput[0]
+	}
+	// var counter1 int = 0
+	// var counter2 int = 0
+	// for {
+	// 	if counter1 >= len(arrayOfMapInput) {
+	// 		break
+	// 	}
+	// 	if calcBMI(arrayOfMapInput[counter1]) > calcBMI(arrayOfMapInput[counter2]) {
+	// 		counter1++
+	// 	}
+	// 	// if calcBMI(arrayOfMapInput[counter1])
+	// }
+	// test return
+	return arrayOfMapInput[0]
+	// return arrayOfMapInput[counter1]
+}
```

Comment:
@nobe4 My understanding is that I somehow have to compare one element with ALL other elements and only if that element is bigger than ALL others it should get returned.
I did start an attempt at a loop, thought of maybe creating a helper function, tried to visualize it with notes and finally with this hard coded element comparison, which obviously is not the goal.
Somehow combining loops sounds good to me, but to piece all of that together is a bit hard right now.
Am I way off or in the right direction?

Reply:
Try the following:

- write a function that takes a list of int and return the biggest.

  ```go
  func largestInt(s []int) int {
    // ???
  }
  ```

- write a function that takes a list of string and return the longest string.

  ```go
  func largestString(s []string) string {
    // ???
  }
  ```

Both functions should have the same pattern, which I can give you if you are too stuck.
Use the same pattern to find the map with the largest BMI.

Reply:
Thank you.
I did find some loop logic example that I tried to change to the loop syntax I previously used.
The `largestInt` does return `0`, then it gets stuck. Might have to give it another try later.
I did try to use the logic in `highestBMI`, also not successfully yet, but maybe closer than before.
Not sure if more hints or just a break is best at this point.

Reply:
I added a comment, but maybe you need to also split the problem into smaller problems?